src/Components/Card/Card.js

```diff
@@ -24,20 +37,30 @@ const Card = ({ content, save }) => {
   const saveHandler = event => {
     event.preventDefault();
     setEdit(false);
-    save({
-      ...content,
+    setCardData({
```

Comment:
you have store outside the component
if you change some card then the store will not know about it and the app might display inconsistent info in different places
you need to modify outer store instead of internal state or additionally to it